src/lib/DataTable/DataTable.svelte

```
@@ -0,0 +1,1357 @@
+  let columnOrder = $state<string[]>(columns.map((c) => c.field));
+
+  // Keep order in sync if the columns prop identity changes (added/removed).
+  $effect(() => {
+    const incoming = columns.map((c) => c.field);
+    const known = new Set(columnOrder);
+    const stillValid = columnOrder.filter((f) => incoming.includes(f));
+    const added = incoming.filter((f) => !known.has(f));
+    if (added.length > 0 || stillValid.length !== columnOrder.length) {
+      columnOrder = [...stillValid, ...added];
+    }
+  });
```

ESLint

[error] $effect is not allowed. Use a different reactive pattern instead.

(no-restricted-syntax)